### PR TITLE
Url generator

### PR DESCRIPTION
**Pull request by pkamps**
What we currently do in PHP to build a URL

```
$myUrl = '/url/path';
// Call by reference
eZURI::transformURI( $myUrl, false, 'full' );
echo "Result: $myUrl";
```

That works fine, it's just a bit tough to use/learn. The 2nd parameter is $ignoreIndexDir which is don't fully understand anyways (never use it). The function name is more complex than it needs to be. The fact that it's called by reference makes it even harder.

This pull requests adds another function to the eZURI class which should be easier to use:

```
$myUrl = '/url/path';
$result = eZURI::build( $myUrl, true );
echo "Result: $result";
```

I added unit tests. It's very unlikely we introduced regression errors: it's a new function.

### CODE REVIEW

File: lib/ezutils/classes/ezuri.php [hunk 1]
```diff
@@ -593,6 +593,29 @@ public static function instance( $uri = false )
...
+     *
+     * @param string $url
+     * @param bool $full
+     * @return string
+     */
+    public static function build( $url, $full = false )
+    {
+        $serverURL = $full ? 'full' : 'relative';
+
+        // Avoid changing origin $url value
```
Comment (peterkeung) [inline lib/ezutils/classes/ezuri.php:607]: "original" not "origin"

**Comment by peterkeung** [comment]
$ignoreIndexDir is used by the "ezroot" template operator.

Certainly I've used "ezroot" a lot in templates, but I can't recall needing "ezroot" behavior in PHP. If that's ever needed, we could use a different function for it to match how that works in templates.

Is _buildPath_ a better function name? I was trying to think of that distinction between a "CMS path" (which ezurl() is, mostly) and a "file system" path (which ezroot() is) but can't come up with something clear.  I have no strong objection to _build_ on its own.

+1 from me

**Comment by pkamps** [comment]
Maybe stick to
eZURI::build
and eventually introduce
eZURI::buildRoot

The path is a little misleading in case you build full URLs.

**Comment by peterkeung** [comment]
👍

**Comment by pkamps** [comment]
It makes a difference if you run the unit tests on the lovestack machine or if travis runs it. The lovestack installation has a proper configuration with additional siteaccesses.
Travis runs the code without that and falls back to the default siteaccess "admin". That siteaccess got removed in this commit: https://github.com/ezsystems/ezpublish-legacy/commit/efafd08a6cd12e54ebb06967b39afed79673042f

**Pull request merged**